src/deluge/gui/views/automation_instrument_clip_view.h

```diff
@@ -144,12 +143,18 @@ class AutomationInstrumentClipView final : public ClipView, public InstrumentCli
 	ModelStackWithAutoParam* getModelStackWithParam(ModelStackWithTimelineCounter* modelStack, InstrumentClip* clip,
 	                                                int32_t paramID = 0xFFFFFFFF,
 	                                                Param::Kind paramKind = Param::Kind::NONE);
+	int32_t getEffectiveLength(ModelStackWithTimelineCounter* modelStack);
+	uint32_t getMiddlePosFromSquare(ModelStackWithTimelineCounter* modelStack, int32_t xDisplay);
```

Comment:
refactored code for getting the value of the middle node in a pad. used for peeking and rendering of display and led indicators